traffic_portal/build/etc/init.d/traffic_portal

```diff
@@ -43,86 +25,50 @@
 # specific language governing permissions and limitations
 # under the License.
 
-# An application name to display in echo text.
-# NAME="Traffic Portal Application"
-# The full path to the directory containing the node and forever binaries.
-# Set the NODE_PATH to the Node.js main node_modules directory.
-# The application startup Javascript file path.
-
-# Process ID file path.
-# Log file path.
-# Forever settings to prevent the application spinning if it fails on launch.
-# MIN_UPTIME="5000"
-# SPIN_SLEEP_TIME="2000"
- 
 NAME="Traffic Portal Application"
 NODE_BIN_DIR="/usr/bin"
 NODE_PATH="/opt/traffic_portal/node_modules"
-FOREVER_BIN_DIR="/opt/traffic_portal/node_modules/forever/bin"
+PM2_BIN_DIR="/opt/traffic_portal/node_modules/pm2/bin"
 APPLICATION_PATH="/opt/traffic_portal/server.js"
 PIDFILE="/var/run/traffic_portal.pid"
 LOGFILE="/var/log/traffic_portal/traffic_portal.log"
-MIN_UPTIME="5000"
-SPIN_SLEEP_TIME="2000"
- 
-# Add node to the path for situations in which the environment is passed.
-PATH=$FOREVER_BIN_DIR:$NODE_BIN_DIR:$PATH
-# Export all environment variables that must be visible for the Node.js
-# application process forked by Forever. It will not see any of the other
-# variables defined in this script.
+RESTART_TIME="2000"
+
+PATH=$PM2_BIN_DIR:$NODE_BIN_DIR:$PATH
 export NODE_PATH=$NODE_PATH
- 
+
 start() {
     echo "Starting $NAME"
-    # We're calling forever directly without using start-stop-daemon for the
-    # sake of simplicity when it comes to environment, and because this way
-    # the script will work whether it is executed directly or via the service
-    # utility.
-    #
-    # The minUptime and spinSleepTime settings stop Forever from thrashing if
-    # the application fails immediately on launch. This is generally necessary to
-    # avoid loading development servers to the point of failure every time
-    # someone makes an error in application initialization code, or bringing down
-    # production servers the same way if a database or other critical service
-    # suddenly becomes inaccessible.
-    #
-    # The pidfile contains the child process pid, not the forever process pid.
-    # We're only using it as a marker for whether or not the process is
-    # running.
-    #
-    # Note that redirecting the output to /dev/null (or anywhere) is necessary
-    # to make this script work if provisioning the service via Chef.
-    forever \
-      --pidFile $PIDFILE \
-      -a \
+    # The pidfile contains the child process pid, and is named of format [pidfile]-[pm2Id].pid
+  pm2 \
+      start "$APPLICATION_PATH" \
+      -p $PIDFILE \
       -l $LOGFILE \
-      --minUptime $MIN_UPTIME \
-      --spinSleepTime $SPIN_SLEEP_TIME \
-      start $APPLICATION_PATH 2>&1 > /dev/null &
+      --restart-delay $RESTART_TIME \
+      --time \
+      --name "$NAME"
     RETVAL=$?
 }
- 
+
 stop() {
-    if [ -f $PIDFILE ]; then
+    echo `pm2 list` | grep -q "$NAME"
+    if [ "$?" -eq "0" ]; then
```

Comment:
nit but <code>echo \`pm2 list\`</code> is equivalent to `pm2 list` and checking the exit code immediately can be more succinctly written as 
```bash
if pm2 list | grep -q "$NAME"; then
```
(add `>/dev/null` if the output to stdout is annoying, won't change the exit code)

Reply:
`-q` on `grep` handles the output.